src/App.tsx

```diff
@@ -1,14 +1,71 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { User } from './types/User';
+import { FilteredOptions } from './types/FilteredOptions';
+
+function handleFilteredTodos(
+  todos: Todo[],
+  optionSelected: FilteredOptions,
+  query: string,
+) {
+  let filteredTodos = [...todos];
+
+  if (query) {
+    filteredTodos = filteredTodos.filter(todo =>
+      todo.title.toLowerCase().includes(query.toLowerCase().trim()),
+    );
+  }
+
+  switch (optionSelected) {
+    case FilteredOptions.active:
+      return filteredTodos.filter(todo => !todo.completed);
+    case FilteredOptions.completed:
+      return filteredTodos.filter(todo => todo.completed);
+
+    default:
+      return filteredTodos;
+  }
+}
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [usersFromServer] = useState<User[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [isModal, setIsModal] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [optionSelected, setOptionSelected] = useState<FilteredOptions>(
+    FilteredOptions.all,
+  );
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(setTodosFromServer)
+      .finally(() => setLoading(false));
+  }, []);
+  function getUserById(userId: number) {
+    return usersFromServer.find(user => user.id === userId) || null;
+  }
+
+  const todos = todosFromServer.map(todo => ({
+    ...todo,
+    user: getUserById(todo.userId),
+  }));
+
+  const preparedTodos = handleFilteredTodos(todos, optionSelected, query);
+  const handleSetOption = (option: FilteredOptions) =>
+    setOptionSelected(option);
```

Comment:
This handler looks redundant as it just passes to the `setOptionSelected` what it gets